src/components/IconButton/IconButton.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { JSXElementConstructor } from 'react';
+
+import styles from './IconButton.module.scss';
+import { IconProps } from '../Icons/Icons.types';
+
+export interface Properties {
+  className?: string;
+  onClick: () => void;
+
+  Icon: JSXElementConstructor<IconProps>;
+  label?: string;
+  size?: string | number;
+  isFilled?: boolean;
+}
+
+export class IconButton extends React.Component<Properties> {
+  handleClick = (event: React.MouseEvent<HTMLButtonElement>): void => {
+    event.stopPropagation();
+    this.props.onClick();
+  };
+
+  render() {
+    return (
+      <button className={`${styles.IconButton} ${this.props.className}`} onClick={this.handleClick}>
```

Comment:
`classNames(styles.IconButton, this.props.className)`, because when concatenating strings if the `className` prop is undefined the resulting class name will be  `IconButton_something undefined`.

Reply:
Yeah, this is one that I couldn't decide on. Feels heavy handed to use a library just to make that check when having `undefined` in the class list isn't really a big deal (other than it looks ugly under the hood). I actually wonder what impact that has on rendering performance :thinking: That'd be interesting to know.

I'll move back to `classNames`. Though, I wonder if there's a better way for us to make this base case simpler given that we do it so often on the root node of a component. My first thought when I see it is usually, "Oh, what are the conditional classes we get on this?" and, of course, in this case there are none. So, it's wasted scanning time.

Reply:
Yeah, it's reasonably minimal. I try to tend towards just not relying on external libs in the cases when it's not necessary (not just for the performance impact).

In any case, I do think if we're going to have this be a common thing across components, adding a class and className prop to the root node every time, then it might be worth seeing if there's a clean way to get that mostly for free every time we create a new component. I'm just not sure if there's a clean way to do it that wouldn't just make things more confusing. We'd have to experiment a bit, I think.